cylc/flow/task_events_mgr.py

```diff
@@ -819,6 +830,31 @@ def _process_message_check(
             self.EVENT_SUBMIT_FAILED, f'{FAIL_MESSAGE_PREFIX}ERR'
         }:
             severity = DEBUG
+
+        # Get job activity log to check whether a message is a repeat.
+        job_activity_log = Path(get_task_job_activity_log(
+            self.workflow,
+            itask.tokens['cycle'],
+            itask.tokens['task'],
+            submit_num
+        ))
+        if not job_activity_log.exists():
+            job_activity_log.parent.mkdir(exist_ok=True, parents=True)
+            job_activity_log.touch()
+
+        # Record received messages in the job activity log.
+        # Do not log it a second time if Polling reveals the same
+        # message has been logged already by reciept method:
+        loggable_message = f'{flag} {itask.identity} {message} {event_time}'
+        if flag == self.FLAG_RECEIVED:
+            with open(os.path.expandvars(job_activity_log), "a") as handle:
+                handle.write(loggable_message + '\n')
+        elif flag == self.FLAG_POLLED:
+            with open(os.path.expandvars(job_activity_log), "r") as handle:
+                log = handle.read()
+                if f'{itask.identity} {message} {event_time}' in log:
+                    return 'quiet'
```

Comment:
Should this be `return None`?

Reply:
I think he's asking if the behaviour around L600 could be controlled by a value of `None` rather than a string

Reply:
Well, yes, it could be, but I think the logic makes more sense if we return an explict (and truthy) value.

Reply:
If this returns `None` then the rest of the previous function would be skipped by the `return` which feels like the behaviour we are after here.

Reply:
Order of events:

* Task submitted (received)
* Task running (received)
* Workflow shut down
* Task hits ETL
* Workflow restarted
* Task failed (polled)

The task failed message is only received once (on restart) so this shouldn't be related?